src/main/java/dk/cngroup/lentils/controller/HintController.java

```diff
@@ -49,43 +59,61 @@ public String deleteHint(@PathVariable("hintId") final Long hintId) {
         Hint hint = hintService.getHint(hintId);
         Long cypherId = hint.getCypherId();
         hintService.deleteById(hintId);
-        return REDIRECT_HINT_LIST + "?cypherId=" + cypherId;
+        return REDIRECT_HINT_LIST + CYPHERID_PARAMETER + cypherId;
     }
 
     @GetMapping(value = "/update/{hintId}")
-    public String addForm(@PathVariable("hintId") final Long hintId, final Model model) {
-        Hint hint = hintService.getHint(hintId);
-        model.addAttribute(TEMPLATE_ATTR_HEADING, "Upravit hint");
-        model.addAttribute(TEMPLATE_ATTR_HINT, hint);
+    public String updateHintForm(@PathVariable("hintId") final Long hintId, final Model model) {
+        HintFormDTO hintFormDto = mapper.map(hintService.getHint(hintId), HintFormDTO.class);
+        model.addAttribute(TEMPLATE_ATTR_HEADING, HEADING_EDIT_HINT);
+        model.addAttribute(TEMPLATE_ATTR_HINT, hintFormDto);
         return VIEW_HINT;
     }
 
+    @PostMapping(value = "/update/{hintId}")
+    public String updateHint(@PathVariable("hintId") final Long hintId,
+                       @Valid @ModelAttribute(TEMPLATE_ATTR_HINT) final HintFormDTO hintFormDto,
+                       final BindingResult bindingResult,
+                       final Model model) {
+        if (bindingResult.hasErrors()) {
+            model.addAttribute(TEMPLATE_ATTR_HEADING, HEADING_EDIT_HINT);
+            model.addAttribute(TEMPLATE_ATTR_HINT, hintFormDto);
+            return VIEW_HINT;
+        }
+
+        Hint hint = null;
+        try {
+            hint = hintService.getHint(hintId);
+        } catch (ResourceNotFoundException e) {
+            throw e;
+        }
+
+        Hint hintToSave = mapper.map(hintFormDto, Hint.class);
+        hintToSave.setHintId(hintId);
```

Comment:
A jo, toho jsem si nevšiml. Tohle není dobrý přístup pro update. vytvořit novou entitu, nastavit jí fieldy a ID a uložit.
Pokud obsahuje řádek v databázi položky, které nemáš ve formuláři (např datum vytvoření entity, soft-delete diskriminátor atd), dost jednodušše se ti může stát, že si to takhle přemažeš.

Lepší je načíst entitu z DB, nasetovat co potřebuješ a uložit.